core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -762,6 +789,103 @@ private[master] class Master(
   }
 
   /**
+   * Handle a request to set the target number of executors for this application.
+   * @return whether the application has previously registered with this Master.
+   */
+  private def handleRequestExecutors(appId: String, requestedTotal: Int): Boolean = {
+    idToApp.get(appId) match {
+      case Some(appInfo) =>
+        logInfo(s"Application $appId requested to set total executors to $requestedTotal.")
+        appInfo.executorLimit = requestedTotal
+
+        // If the application raises the executor limit, then we can launch new executors.
+        // If there are previously blacklisted workers, then we can launch these new executors
+        // by unblacklisting a subset of these workers. For more detail, see `handleKillExecutors`.
+        if (appInfo.desc.coresPerExecutor.isEmpty) {
```

Comment:
I think we may not need to execute removeFromBlacklist if the application requests less resources than what it has been assigned? (it doesn't change the correctness of the program, since we have this line https://github.com/apache/spark/pull/7532/files#diff-29dffdccd5a7f4c8b496c293e87c8668R598